polysemy-plugin/test/MultipleVarsSpec.hs

```diff
@@ -0,0 +1,28 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -fplugin=Polysemy.Plugin #-}
+
+module MultipleVarsSpec where
+
+import Polysemy
+
+import Test.Hspec
+
+data TaggedState k s m a where
+  TaggedGet :: forall k s m. TaggedState k s m s
+  TaggedPut :: forall k s m. s -> TaggedState k s m ()
+
+makeSem ''TaggedState
+
+test :: Members '[
+          TaggedState Char Int
+        , TaggedState Bool Int
+        ] r
+      => Sem r ()
+test = taggedPut @Bool 10
+
+
+spec :: Spec
+spec = describe "Using multiple, but ununifiable instances\
+               \ of the same effect" $ do
+  it "should get disambiguated and compile!" $ do
+    True `shouldBe` True
```

Comment:
Can you add a test that ensures it also works without an explicit `Given`? ie

```haskell
flip shouldBe blah . run . runTagged @Bool . runTagged @Char $ do
  taggedPut @Bool 10
```

Reply:
Also I'd like to see actually running `test`, to ensure the tagged effects end up in the right place. Right now we just give evidence by fiat, which will happily compile even if it _does the wrong thing_ :scream:

Reply:
Just tested it. Works fine; I'll commit the update to the test together with any other changes you'd like.